lib/ansible/modules/apt.py

```diff
@@ -634,8 +634,11 @@ def expand_pkgspec_from_fnmatches(m, pkgspec, cache):
 
             pkgname_pattern, version_cmp, version = package_split(pkgspec_pattern)
 
+            if pkgname_pattern.startswith(('?', '~', '(?', '(~')):
+                # this is an apt-pattern (https://manpages.debian.org/bookworm/apt/apt-patterns.7.en.html)
+                new_pkgspec.append(pkgspec_pattern)
             # note that none of these chars is allowed in a (debian) pkgname
-            if frozenset('*?[]!').intersection(pkgname_pattern):
+            elif frozenset('*?[]!').intersection(pkgname_pattern):
```

Comment:
Please create a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment). See [this fragment](https://github.com/ansible/ansible/blob/c2f528b7686c9573aa8493357219e0bbe4416dd3/changelogs/fragments/reboot-change-default-boot-command.yaml) as an example. Also, add a test for this change.

Reply:
Done. There was already a unit for the enclosing function `expand_pkgspec_from_fnmatches(m, pkgspec, cache)` to which I added another parameter set.